### ISSUE TITLE
Project submission .ipynb files

### ISSUE THREAD

**Issue by jwang727** [tomncooper/CSC8101-Documentation#19]
When submitting python notebook files to NESS can we submit them as .ipynb.json files (what you get if you download the file from Jupyter) or do they need to be converted to something more readable? If so how do we do that?

I am guessing we shouldn't just directly scp the .ipynb files from the virtual machine and should at least convert them to .ipynb.json?

**Comment by tomncooper** [comment]
You should export them as Python script files (.py). With the notebook open there should be an option under File -> Download As -> Python (.py)

**Comment by jwang727** [comment]
If I do that I get a py.html file. What's your recommended way to convert a py.html file into a .py file, if we have to do that?

**Comment by PetoMichalak** [comment]
Try to export the python script from your iPython notebook as shown on the picture below:

![Exporting python script](https://github.com/tomncooper/CSC8101-Documentation/blob/master/spark/images/export_python_from_notebook.png?raw=true)

This should download a python file in a format: <notebook_name>.py; which you should submit to the ness.

**Comment by jwang727** [comment]
That's exactly what I am doing but I get a .py.html file.

**Comment by PetoMichalak** [comment]
I've looked into this again and I see there is a different behaviour depending on the browser one is using. I've tested the described approach on Chrome and Firefox with success, however when using Safari I did get <notebook_name>.py.html file name. Once I examined the file in an editor this was a pure python script (no html tags included). 

**Things to try**:
* Use a different browser to download your final solution,
or
* Examine the downloaded .html file and if there are no [html tags](https://www.w3schools.com/TAGs/) present, remove the additional '.html' extension from the file before submitting it.

Let us know if this helped!
